File: Android_App_Java_Gucci_Store/app/src/main/java/com/example/dhura_mistry_project2/DetailActivity.java
package com.example.dhura_mistry_project2;

import static android.content.ContentValues.TAG;

import static java.lang.Integer.parseInt;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

import com.google.android.gms.tasks.OnCompleteListener;
import com.google.android.gms.tasks.Task;
import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.firestore.DocumentReference;
import com.google.firebase.firestore.FirebaseFirestore;
import com.squareup.picasso.Picasso;

import org.w3c.dom.Text;

import java.text.SimpleDateFormat;
import java.util.ArrayList;
import java.util.Calendar;
import java.util.HashMap;

public class DetailActivity extends AppCompatActivity  {

    ArrayList<CartItems> list = new ArrayList<>() ;
    int totalQuantity;
    int totalPrice;
    TextView itemName,itemPrice,quantity,itemDescription;
    ImageView itemImage,addItem,removeItem,btnDetailCart,btnDetailHome;
    Button btnCheckout,btnDetailAddToCart;
    String name,image,price,description;
    FirebaseFirestore firestore;
    FirebaseAuth auth;

    Products Products = null;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_detail);

        firestore = FirebaseFirestore.getInstance();
        auth = FirebaseAuth.getInstance();

        name = getIntent().getStringExtra("Name");
        image = getIntent().getStringExtra("Image");
        price = getIntent().getStringExtra("Price");
        description = getIntent().getStringExtra("Description");
        totalQuantity = 1;

        Log.i(TAG, "onCreate: "+name+image+price);

        itemName = findViewById(R.id.nameDetailo);
        itemPrice = findViewById(R.id.priceDetail);
        itemImage = findViewById(R.id.imageDetail);
        itemDescription = findViewById(R.id.itemDescription);
        btnCheckout = findViewById(R.id.btnCheckout);
        btnDetailCart = findViewById(R.id.btnDetailCart);
        btnDetailHome = findViewById(R.id.btnDetailHome);
        btnDetailAddToCart = findViewById(R.id.btnDetailAddToCart);
        quantity = findViewById(R.id.Quantity);
        addItem = findViewById(R.id.addItem);
        removeItem = findViewById(R.id.removeItem);

        itemName.setText(name);
        itemPrice.setText("$"+price);
        itemDescription.setText(description);
        Picasso.get().load(image).into(itemImage);

        totalPrice = parseInt(price) * totalQuantity;

        btnDetailCart.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(DetailActivity.this,CartActivity.class);
                startActivity(intent);
            }
        });

        btnDetailHome.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(DetailActivity.this,ProductsActivity.class);
                startActivity(intent);
            }
        });

        btnDetailAddToCart.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
            addedToCart();
            }
        });

        addItem.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
             if(totalQuantity <10){
                 totalQuantity++;
                 quantity.setText(String.valueOf(totalQuantity));
                 totalPrice = parseInt(price) * totalQuantity;
             }
            }
        });

        removeItem.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if(totalQuantity <10){
                    totalQuantity--;
                    quantity.setText(String.valueOf(totalQuantity));
                    totalPrice = parseInt(price) * totalQuantity;
                }
            }
        });


        Intent i = new Intent(this,CheckoutActivity.class);

        btnCheckout.setOnClickListener(new View.OnClickListener() {

            @Override
            public void onClick(View v) {
                startActivity(i);
//                    list.add(new CartItems(image,name,price));
            }
        });
//        itemImage.setImageResource(image);
    }

    private void addedToCart() {
        final HashMap<String,Object> cartMap = new HashMap<>();

        cartMap.put("itemName",name);
        cartMap.put("itemPrice",price);
        cartMap.put("itemImage",image);
        cartMap.put("quantity",totalQuantity);
        cartMap.put("totalPrice",totalPrice);

        firestore.collection("AddToCart").document(auth.getCurrentUser().getUid()).collection("CurrentUser").add(cartMap).addOnCompleteListener(new OnCompleteListener<DocumentReference>() {
            @Override
            public void onComplete(@NonNull Task<DocumentReference> task) {
                Toast.makeText(DetailActivity.this,"Added to a cart",Toast.LENGTH_SHORT).show();

            }
        });
    }
}

File: Kotlin_App_1/app/src/main/java/com/example/group1_project1/DetailActivity.kt
package com.example.group1_project1


import android.content.Intent
import android.os.Bundle
import android.widget.Button
import android.widget.ImageView
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.bumptech.glide.Glide
import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.database.DatabaseReference
import com.google.firebase.database.FirebaseDatabase
import com.google.firebase.firestore.FirebaseFirestore
import com.google.firebase.storage.FirebaseStorage
import com.google.firebase.storage.StorageReference
import java.lang.Integer.parseInt


class DetailActivity : AppCompatActivity() {

    var auth: FirebaseAuth? = null
    var totalQuantity: Int = 0
    var totalPrice: Int = 0

    var Name: String = ""
    var Price: String = ""
    var Image: String = ""
    var Desc: String = ""
    private lateinit var database: DatabaseReference


    override fun onCreate(savedInstanceState: Bundle?) {
        
        auth = FirebaseAuth.getInstance();

        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_detail)

        val itemName: TextView = findViewById(R.id.nameDetailo);
        val itemPrice: TextView = findViewById(R.id.priceDetail);
        val itemImage : ImageView = findViewById(R.id.imageDetail);
        val itemDesc : TextView =  findViewById(R.id.itemDescription);
        val btnCheckout : Button = findViewById(R.id.btnCheckout);
        val btnAddToCart : Button = findViewById(R.id.btnDetailAddToCart);
        val btnDetailCart: ImageView = findViewById<ImageView>(R.id.btnDetailCart)
        val btnDetailHome: ImageView = findViewById<ImageView>(R.id.btnDetailHome)
        val btnDetailAddToCart : Button = findViewById<Button>(R.id.btnDetailAddToCart)
        val quantity: TextView = findViewById<TextView>(R.id.Quantity)
        val addItem: ImageView = findViewById<ImageView>(R.id.addItem)
        val removeItem :ImageView = findViewById<ImageView>(R.id.removeItem)


        val bundle : Bundle?= intent.extras
        Name = bundle!!.getString("name").toString()
        Price = bundle.getString("price").toString()
        Desc = bundle.getString("desc").toString()
        Image = bundle.getString("image").toString()
        totalQuantity = 1;


        itemName.text = Name;
        itemDesc.text = Desc;
        itemPrice.text = "$" + Price;
        val storRef: StorageReference = FirebaseStorage.getInstance().getReferenceFromUrl(Image.toString())
        Glide.with(itemImage.context).load(storRef).into(itemImage)
        totalPrice = parseInt(Price) * totalQuantity;

        btnDetailCart.setOnClickListener {
            val intent = Intent(this@DetailActivity, CartActivity::class.java)
            startActivity(intent)
        }

        btnDetailHome.setOnClickListener {
            val intent = Intent(this@DetailActivity, ProductActivity::class.java)
            startActivity(intent)
        }

        addItem.setOnClickListener {
            if (totalQuantity < 10) {
                totalQuantity++
                quantity.text = totalQuantity.toString()
                totalPrice = Price.toInt() * totalQuantity
            }
        }

        removeItem.setOnClickListener {
            if (totalQuantity < 10) {
                totalQuantity--
                quantity.text = totalQuantity.toString()
                totalPrice = Price.toInt() * totalQuantity
            }
        }

        btnCheckout.setOnClickListener {
            val intent = Intent(this@DetailActivity, CheckoutActivity::class.java);
            startActivity(intent)
        }

        btnAddToCart.setOnClickListener {
            addedToCart()
        }

    }

    private fun addedToCart() {
//        val cartMap: HashMap<String, Any> = HashMap()
//        cartMap["itemName"] = Name
//        cartMap["itemPrice"] = Price
//        cartMap["itemImage"] = Image
//        cartMap["quantity"] = totalQuantity
//        cartMap["totalPrice"] = totalPrice

        database = FirebaseDatabase.getInstance().getReference("Cart")
        val Cart = CartItems(Image,Name,Price,totalQuantity,totalPrice)
        database.child(auth!!.currentUser!!.uid).push().setValue(Cart).addOnSuccessListener {

            Toast.makeText(this@DetailActivity, "Added to a cart", Toast.LENGTH_SHORT).show()

        }.addOnFailureListener{
            Toast.makeText(this@DetailActivity, "Error could not add to a cart", Toast.LENGTH_SHORT).show()
        };


//        firestore!!.collection("AddToCart").document(auth!!.currentUser!!.uid)
//            .collection("CurrentUser").add(cartMap).addOnCompleteListener {
//                Toast.makeText(
//                    this@DetailActivity,
//                    "Added to a cart",
//                    Toast.LENGTH_SHORT
//                ).show()
//            }
    }


}